CMakeLists.txt

```
@@ -96,13 +101,13 @@ set_target_properties(lib_kmeans
         PROPERTIES
         VERSION 0.0.1
         SOVERSION 0.0.1
```

```suggestion
        VERSION ${libkmeans_VERSION}
        SOVERSION ${libkmeans_VERSION_MAJOR}
```
Never set `SOVERSION == VERSION`. That's a middle-finger to packagers

---

I have never written cmake for packaging. I just set these out of convenience.

---

SOVERSION: Version of the cross-*compatible* libraries. I.e. when you link to a library it links to the specific SOVERSION used at build time. If you update a package, but the SOVERSION is identical, it will continue to work. But if you set it like that, you are indicating, no version is compatible with one-another.